tests/test_datetime/test_datetime_subtraction.py

```diff
@@ -216,7 +216,7 @@ def test_raises_error_when_nan_in_transform():
     "unit, expected",
     [
         ("D", [31, 61, 183]),
-        ("M", [1.018501, 2.004148, 6.012444]),
+        ("M", [1.0, 1.967741935483871, 5.903225806451613]),
```

Comment:
why are the values changing?

Reply:
I need some time to deep dive into this, it is strange indeed. Might be something changed with subtraction between times or something changed with divisions of time deltas

Reply:
Most likely the latter as the test with days passes with no problem. Might as well be a bug with the new version of pandas.

Reply:
If you want, we could also fix this by reverting the change in the tests and by changing [line 321 of datetime /datetime_subtraction.py.py](https://github.com/feature-engine/feature_engine/blob/main/feature_engine/datetime/datetime_subtraction.py#L321) as follows

```diff
    def _sub(self, dt_df: pd.DataFrame):
        """make datetime subtraction"""
        new_df = pd.DataFrame()
        for reference in self.reference_:
            new_varnames = [f"{var}_sub_{reference}" for var in self.variables_]
            new_df[new_varnames] = (
                dt_df[self.variables_]
                .sub(dt_df[reference], axis=0)
-                .apply(lambda s: s / np.timedelta64(1, self.output_unit))
+                .div(np.timedelta64(1, self.output_unit).astype("timedelta64[ns]"))
            )
```

Reply:
yes, `.div` looks more pro than the lambda function.

Awesome work, thanks for digging this out!

It would be great if you can take care of this, then we are good to merge :)

Reply:
Thanks! And thanks for your review comments!

I have made the change from the previous comment.

Let me know if you have other comments.

Before merging, it would be great to run tests with pandas 1.5.x just to double check that changes are backward compatible.

Reply:
I have tried downgrading to pandas 1.5.3 in my laptop and running the tests. Everything passed for me.

Reply:
Amazing! Thank you!